src/main/java/com/hermesworld/ais/galapagos/topics/controller/TopicController.java

```diff
@@ -99,38 +96,28 @@ public TopicController(ValidatingTopicService topicService, KafkaClusters kafkaE
         }
     }
 
-    @GetMapping(value = "/api/applicationsmetadata/{envId}/{topicName}", produces = MediaType.APPLICATION_JSON_VALUE)
-    public List<KnownApplicationDto> getAvailableApplicationMetadata(@PathVariable String envId,
-            @PathVariable String topicName,
-            @RequestParam(name = "topicOwnerId", required = false) String topicOwnerId) {
-        TopicMetadata topic = topicService.getTopic(envId, topicName).orElseThrow(notFound);
-        List<String> producerIds = topic.getProducers();
-
-        return applicationsService.getAllApplicationMetadata(envId).stream().map(ApplicationMetadata::getApplicationId)
-                .filter(id -> !id.equals(topicOwnerId))
-                .filter(idsWithoutOwnerid -> !producerIds.contains(idsWithoutOwnerid))
-                .map(id -> toKnownAppDto(applicationsService.getKnownApplication(id)
-                        .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND))))
-                .collect(Collectors.toList());
-    }
-
-    @PostMapping(value = "/api/producers/{environmentId}", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public CompletableFuture<Void> addProducerToTopic(@PathVariable String environmentId,
-            @RequestBody AddProducerDto producers) {
-        if (producers.getProducers().stream().anyMatch(appId -> !applicationsService.isUserAuthorizedFor(appId))) {
+    @PostMapping(value = "/api/producers/{environmentId}/{topicName:.+}", consumes = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
We should better remove this `:.+`. This is not always used, and it is not required, as e.g. slashes are not allowed in topic names anyway. Please just use `{topicName}` (and yes, we must also remove this from other endpoints in the future).